File: app/src/main/kotlin/com/spacex/ui/errorView/ErrorView.java
package com.spacex.ui.errorView;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.databinding.DataBindingUtil;
import com.spacex.ui.R;
import com.spacex.ui.databinding.ErrorViewBinding;

public class ErrorView extends ConstraintLayout {

  private OnRefreshListener onRefreshListener;

  public ErrorView(@NonNull Context context) {
    this(context, null);
  }

  public ErrorView(@NonNull Context context, @Nullable AttributeSet attrs) {
    this(context, attrs, 0);
  }

  public ErrorView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
    init(context, attrs);
  }

  private void init(@NonNull Context context, @Nullable AttributeSet attributeSet) {
    final ErrorViewBinding viewBinding =
        DataBindingUtil.inflate(LayoutInflater.from(context), R.layout.error_view, this, true);
    viewBinding.retry.setOnClickListener(
        view -> {
          if (onRefreshListener != null) {
            onRefreshListener.onRefresh();
          }
        });
  }

  public void setOnRetryListener(@NonNull OnRefreshListener onRefreshListener) {
    this.onRefreshListener = onRefreshListener;
  }

  public interface OnRefreshListener {

    void onRefresh();
  }
}

File: app/src/main/kotlin/com/spacex/ui/customViews/ErrorView.kt
package com.spacex.ui.customViews

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.databinding.DataBindingUtil
import com.spacex.ui.R
import com.spacex.ui.databinding.ErrorViewBinding

class ErrorView @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null,
        defStyleAttr: Int = 0) : ConstraintLayout(context, attrs, defStyleAttr) {

    private var onRefreshListener: OnRefreshListener? = null

    init {
        init(context)
    }

    private fun init(context: Context) {
        val viewBinding = DataBindingUtil.inflate<ErrorViewBinding>(
                LayoutInflater.from(context), R.layout.error_view, this, true)
        viewBinding.retry.setOnClickListener {
            if (onRefreshListener != null) {
                onRefreshListener!!.onRefresh()
            }
        }
    }

    fun setOnRetryListener(onRefreshListener: OnRefreshListener) {
        this.onRefreshListener = onRefreshListener
    }

    interface OnRefreshListener {
        fun onRefresh()
    }
}
